File: src/main/java/lk/ijse/cmjd/researchtracker/milestone/Milestone.java
package lk.ijse.cmjd.researchtracker.milestone;

import jakarta.persistence.*;
import lk.ijse.cmjd.researchtracker.project.Project;
import lk.ijse.cmjd.researchtracker.user.User;
import lombok.*;
import java.time.LocalDate;
import java.util.UUID;

@Entity
@Table(name = "milestones")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Milestone {

    @Id
    private String id = UUID.randomUUID().toString();

    @ManyToOne
    @JoinColumn(name = "project_id")
    private Project project;

    @Column(nullable = false)
    private String title;

    private String description;

    private LocalDate dueDate;

    private boolean isCompleted = false;

    @ManyToOne
    @JoinColumn(name = "created_by")
    private User createdBy;
}

File: frontend/src/types/Milestone.ts
import { User } from "./User";
import { Project } from "./Project";

export interface Milestone {
    id: string;
    project: Project;
    title: string;
    description: string;
    dueDate: string;
    isCompleted: boolean;
    createdBy: User;
    createdAt: string;
}